app/pages/api/ask.ts

```diff
@@ -93,13 +93,13 @@ async function createContext(question: string) {
   return { context, pageChunks };
 }
 
-async function getPageResults(question: string) {
+async function getPageResults(question: string): Promise<PageChunk[]> {
   const embedding = await createEmbedding(question);
   const pageChunks = await searchPages(embedding);
-  const uniqueLinks = Array.from(
-    new Set<string>(pageChunks.map((chunk) => chunk.url))
+  const uniqueChunks = Object.fromEntries(
+    pageChunks.map((chunk) => [chunk.url, chunk])
   );
-  return uniqueLinks;
+  return Object.values(uniqueChunks);
```

Comment:
I think you could remove the fromEntries/values calls here unless you have big plans

Reply:
What do you suggest?
 
I think the purpose was merging on the key (i.e. returning only one entry for each unique URL). I don't really have a strong opinion on how this should work, but in the future I'm thinking it could be unique-ified on the client side either way so that the backend isn't unnecessarily withholding information. I just didn't want to change too much.